net/http/authn/authn.go

```diff
@@ -60,6 +66,13 @@ func (a *API) Authenticate(req *http.Request) (*http.Request, error) {
 	return req.WithContext(ctx), nil
 }
 
+func (a *API) certAuthn(req *http.Request) (*pkix.Name, error) {
```

Comment:
Prob cleaner to have this function return the new request after wrapping the context, then there's no need for an error value (which will only be dropped anyway).